packages/cli/src/commands/onboard/discovery/index.ts

```diff
@@ -0,0 +1,141 @@
+/*
+ * Copyright 2023 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import fs from 'fs-extra';
+import yaml from 'yaml';
+import inquirer from 'inquirer';
+import chalk from 'chalk';
+import { loadCliConfig } from '../../../lib/config';
+import { updateConfigFile } from '../config';
+import { APP_CONFIG_FILE, DISCOVERED_ENTITIES_FILE } from '../files';
+import { Discovery } from './Discovery';
+import { BasicRepositoryAnalyzer } from './analyzers/BasicRepositoryAnalyzer';
+import { PackageJsonAnalyzer } from './analyzers/PackageJsonAnalyzer';
+import { GithubDiscoveryProvider } from './providers/github/GithubDiscoveryProvider';
+import { GitlabDiscoveryProvider } from './providers/gitlab/GitlabDiscoveryProvider';
+import { GitHubAnswers, GitLabAnswers } from '../auth';
+import { Task } from '../../../lib/tasks';
+
+export async function discover(providerInfo?: {
+  provider: string;
+  answers: GitHubAnswers | GitLabAnswers;
+}) {
+  Task.log(`
+    Would you like to scan for - and create - Software Catalog entities?
+
+    You will need to select which SCM (Source Code Management) provider you are using, 
+    and then which repository or organization you want to scan.
+
+    This will generate a new file in the root of your project containing discovered entities,
```

Comment:
possible risk of confusion of the word "project" here - can it be read as expecting a file to be created in the remote repo being scanned? Not sure.